cmd/agent/main.go

```diff
@@ -42,14 +42,14 @@ func main() {
 		os.Exit(0)
 	}
 
-	logFilePath, err := filepath.Abs("cirrus-agent.log")
-	if err != nil {
-		log.Printf("Failed to get absolute log file path: %v", err)
-		logFilePath = "cirrus-agent.log"
-	}
-	logFile, err := os.OpenFile(logFilePath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0660)
+	logFilePath := filepath.Join(os.TempDir(), fmt.Sprintf("cirrus-agent-%d.log", *taskIdPtr))
+	defer uploadAgentLogs(logFilePath, *taskIdPtr, *clientTokenPtr)
```

Comment:
Looks like this `defer` will be executed after `defer conn.Close()` below and this might cause the log uploading to always fail.

Reply:
Indeed! Good catch! Fixed in 1ddb2f6964e38381f433ad87bbe821b30f290604 by not trying to use so much defers.